math/fibonacci_sum.cpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file
+ * @brief An algorithm to calculate the sum of Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$
+ * @details An algorithm to calculate the sum of Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$ where \f$\mathrm{F}(i)\f$
+ * denotes the i-th Fibonacci Number . Note that F(0) = 0 and F(1) = 1.
+ * The value of the sum is calculated using matrix exponentiation.
+ * @source https://stackoverflow.com/questions/4357223/finding-the-sum-of-fibonacci-numbers
+ * @author [Sarthak Sahu](https://github.com/SarthakSahu1009)
+ */
+
+#include <iostream> /// for std::cin and std::cout
+#include <cassert>  /// for assert
+#include <vector>   /// for std::vector
+
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+    /**
+     * @namespace fibonacci_sum
+     * @brief Functions for the sum of the Fibonacci Sequence: \f$\mathrm{F}(n) + \mathrm{F}(n+1) + .. + \mathrm{F}(m)\f$
+     */
+    namespace fibonacci_sum {
+
+        /**
+         * Function to multiply two matrices
+         * @param T matrix 1
+         * @param A martix 2
+         * @returns resultant matrix
+         */
+        std::vector<std::vector<int> > multiply(std::vector<std::vector<int> > T, std::vector<std::vector<int> > A) {
+
+            std::vector<std::vector<int> > result(2,std::vector<int>(2));
+
+            // multiplying matrices
+            for(int i=0;i<2;i++) {
+                for(int j=0;j<2;j++) {
+                    result[i][j]=0;
+                    for(int k=0;k<2;k++) {
+                        result[i][j]=(result[i][j]+(T[i][k]*A[k][j]));
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        /**
+         * Function to compute A^n where A is a matrix.
+         * @param T matrix
+         * @param ex power
+         * @returns resultant matrix
+         */
+        std::vector<std::vector<int> > power(std::vector<std::vector<int> > T, int ex) {
+
+            std::vector<std::vector<int> > A{{1,1},{1,0}};
+            if(ex == 0 || ex == 1) {
+                return T;
+            }
+
+            T = power(T,ex/2);
+            T = multiply(T,T);
+            if(ex&1) {
+                T = multiply(T,A);
+            }
+            return T;
+        }
+
+        /**
+         * Function to compute sum of fibonacci sequence from 0 to n.
+         * @param n number
+         * @returns int ans, the sum of sequence
+         */
+        int result(int n) {
```

Comment:
the input n MUST be positive. Is there a limit how big it can be?
Also, will the result be sufficient in the default `short int` or should it rather be `uint64_t`?
```suggestion
        int result(uint8_t n) {
```

Reply:
Changed.